### ISSUE TITLE
Intellij Reporting `open` method is 

### ISSUE THREAD

**Issue by pluttrell** [ajoberstar/gradle-git#240]
Intellij (2017.1 EAP) is reporting that the `Grgit.open()` method is deprecated. What should we use instead?

<img width="511" alt="screen shot 2017-03-06 at 11 28 29 am" src="https://cloud.githubusercontent.com/assets/4094051/23626116/8cc20966-0260-11e7-9000-d20cd40ca72b.png">

I'm using `id "org.ajoberstar.grgit" version "1.6.0"`, which I understand to be the latest version.

For reference, I chose this `open` method based on this [wiki article](https://github.com/ajoberstar/gradle-git/wiki/Release%20Plugins#configuration).

In case it matters, my use case is just to determine if my repo is clean/dirty, and I'm using `repo.status().clean` to determine such.

**Comment by ajoberstar** [comment]
The deprecated version of `open` is the one taking a single un-named argument. The documentation of [OpenOp](http://ajoberstar.org/grgit/docs/groovydoc/org/ajoberstar/grgit/operation/OpenOp.html) shows the new methods, which take named arguments depending on how exactly you want to search for the git directory.

I've also updated that wiki page to use a non-deprecated call.

**Comment by pluttrell** [comment]
@ajoberstar Thanks for getting back to me. I tried the named versions with the latest version (1.7.2) of the plugin and the latest version of Intellij (2017.2) and still have the same problem.

<img width="672" alt="screen shot 2017-07-19 at 5 03 22 pm" src="https://user-images.githubusercontent.com/4094051/28394802-9117c5f6-6ca4-11e7-8189-35cef705647d.png">

In addition to `dir:`, I've tried `currentDir:` and see the same thing.

For reference here's the full `build.gradle` that demonstrates what I'm seeing:

```
buildscript {
    repositories {
        jcenter()
    }
}

plugins {
    id "org.ajoberstar.grgit" version "1.7.2"
}

apply plugin: 'java'

repositories {
    jcenter()
}

def repo = org.ajoberstar.grgit.Grgit.open(dir: '.')

task validateGitRepoClean {
    doFirst {
        assert repo.status().clean : 'Git repo found to contain uncommitted changes. A clean repo is required for this task'
    }
}
```

**Comment by ajoberstar** [comment]
This is an artifact of the way Grgit uses the Groovy MOP. The only concrete methods are the deprecated ones, so Intellij reports them that way.

This will be fixed in 2.0, but your usage is correct.
